src/MakeCTe.php

```diff
@@ -619,32 +498,31 @@ public function monta()
             }
             $this->dom->appChild($this->infCte, $this->compl, 'Falta tag "infCte"');
         }
-        $this->dom->appChild($this->emit, $this->enderEmit, 'Falta tag "emit"');
+        // inclui o Node enderEmit dentro do emit antes da tag CRT
+        $node = $this->emit->getElementsByTagName("CRT")->item(0);
+        $this->emit->insertBefore($this->enderEmit, $node);
+
         $this->dom->appChild($this->infCte, $this->emit, 'Falta tag "infCte"');
-        if ($this->rem != '') {
+        if (!empty($this->rem)) {
             $this->dom->appChild($this->infCte, $this->rem, 'Falta tag "infCte"');
         }
-        if ($this->exped != '') {
+        if (!empty($this->exped)) {
             $this->dom->appChild($this->infCte, $this->exped, 'Falta tag "infCte"');
         }
-        if ($this->receb != '') {
+        if (!empty($this->receb)) {
             $this->dom->appChild($this->infCte, $this->receb, 'Falta tag "infCte"');
         }
-        if ($this->dest != '') {
+        if (!empty($this->dest)) {
             $this->dom->appChild($this->infCte, $this->dest, 'Falta tag "infCte"');
         }
         foreach ($this->comp as $comp) {
             $this->dom->appChild($this->vPrest, $comp, 'Falta tag "vPrest"');
         }
         $this->dom->appChild($this->infCte, $this->vPrest, 'Falta tag "infCte"');
         $this->dom->appChild($this->infCte, $this->imp, 'Falta tag "imp"');
-        if ($this->infCteComp != '') { // Caso seja um CTe tipo complemento de valores
+        if ($this->tpCTe == 3 and !empty($this->infCteComp)) { // Caso seja um CTe tipo complemento de valores
             $this->dom->appChild($this->infCte, $this->infCteComp, 'Falta tag "infCteComp"');
-        }
-        if ($this->infCteAnu != '') { // Caso seja um CTe tipo anulação
-            $this->dom->appChild($this->infCte, $this->infCteAnu, 'Falta tag "infCteAnu"');
-        }
-        if ($this->infCTeNorm != '') { // Caso seja um CTe tipo normal
+        } elseif (in_array($this->tpCTe, [0, 1]) and !empty($this->infCTeNorm)) { // Caso seja um CTe tipo normal
```

Comment:
`} elseif (in_array($this->tpCTe, [0, 3]) and !empty($this->infCTeNorm)) { // Caso seja um CTe tipo normal`